latch_config/config.py

```diff
@@ -51,17 +70,92 @@ class LoggingMode(str, Enum):
 T = TypeVar("T")
 
 
-def read_config(x: type[T], env_prefix: str = "") -> T:
+class _IsDataclass(Protocol):
```

Comment:
this is from pyright's standard library types, right?

Reply:
i have no idea